feat: Update Managing branches page CY-6043 (#1483)
codacy/docs: docs/repositories-configure/managing-branches.md

# Managing branches

Codacy automatically analyzes the default branch of your repository (typically `master` or `main` as configured on your Git provider), and also supports analyzing multiple branches.

To change the default branch of your repository or enable analysis on other branches:

1.  Open your repository **Settings**, tab **Branches**.

    ![Managing branches](images/managing-branches.png)

1.  To enable a new branch, toggle the corresponding switch in the column **Analyze**.

    Enabling a new branch triggers an initial analysis of that branch and the analysis results and information for that branch will become available after the analysis is complete.

1.  To change the default branch on Codacy, click the corresponding **Make default** link in the column **Default**.

    Codacy displays information for the default branch on dashboards and the repositories list by default, and we recommend that you set the default branch on Codacy to match the default branch of your Git repository.

>     <!--TODO Check if it's important to mention this-->
>     !!! note
>         To set a branch as the default branch, you must first enable that branch.

Reviewer: I think it is a good info yes but I would rephrase to something like: 
```suggestion
    !!! note
        You can only set a default branch an already analyzed branch.
```
or something
What you think?

Author: Great suggestion, thanks!

1.  To change the default branch on Codacy, click the corresponding **Make default** link in the column **Default**.

    Codacy displays information for the default branch on dashboards and the repositories list by default, and we recommend that you set the default branch on Codacy to match the default branch of your Git repository.

    !!! note
        You can only set as the default branch an already analyzed branch.
{% else %}
Codacy automatically triggers analysis on the main branch of your repository (typically `master` or `main` as configured on your Git provider), and also supports analyzing multiple branches.

To change the main branch of your repository or enable analysis on other branches, open your repository **Settings**, tab **Branches**. Enabling a new branch triggers an initial analysis of that branch and the analysis results and information for that branch will become available after the analysis is complete.

!!! note
    **If you're using Codacy Self-hosted** you can also configure Codacy to automatically enable all new branches that are pushed to the repository.

> 

Reviewer: If it's self-hosted we will show the other version below right? Or the ideia is just to add self-hosted version bellow?

Author: I was taking a shortcut :sweat_smile: and trying to solve two issues on the same pull request, but I guess it's better to have separate pull requests:

- I opened the dedicated pull request https://github.com/codacy/docs/pull/1486 to add the missing Self-hosted option **Auto-enable new branches**, and I'm reverting those changes on this pull request
- The scope of this pull request is to refactor the docs to take into account the migrated **Branches** tab, while keeping the old content for Self-hosted

I'm hoping that this makes everything clearer. :pray:

# Managing branches

{% if not extra.self_hosted %}
Codacy automatically analyzes the default branch of your repository (typically `master` or `main` as configured on your Git provider), and also supports analyzing multiple branches.

To change the default branch of your repository or enable analysis on other branches:

1.  Open your repository **Settings**, tab **Branches**.

    ![Managing branches](images/managing-branches.png)

1.  To enable analysis for a new branch, toggle the corresponding switch in the column **Analyze**.

    Enabling a new branch triggers an initial analysis of that branch and the analysis results and information for that branch will become available after the analysis is complete.

1.  To change the default branch on Codacy, click the corresponding **Make default** link that appears when you hover the column **Default**.

    Codacy displays information for the default branch on dashboards and the repositories list by default, and we recommend that you set the default branch on Codacy to match the default branch of your Git repository.

    !!! note

>         You can only set as default branch an already enabled branch.

Author: @alerizzo After reading [your clarification](https://github.com/codacy/codacy-website/pull/286#discussion_r1000297455) I realize it's perhaps a good idea to make it more explicit that users have to wait until a branch is actually **analyzed** before they can set it as default, do you agree?

```suggestion
        You can only set as default branch an already enabled and analyzed branch.
```

Reviewer 2: We can, although I think the API is not waiting until the end of the first analysis. Only until the enable change was done.

Author: Thanks for clarifying! :bow: We'll leave it as is.
